### ISSUE TITLE
nuxt generate fails 3.12.3 vs 3.12.2

### ISSUE THREAD

**Issue by PhE** [nuxt/nuxt#28148]
### Environment

- Operating System: Linux
- Node Version:     v18.19.1
- Nuxt Version:     3.12.3
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  pnpm@9.3.0
- Builder:          -
- User Config:      compatibilityDate, devtools
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

Start with a fresh nuxt project:
```shell
bunx nuxi@latest init nuxt-generate
cd nuxt-generate
bun install
bun run generate
npx serve .output/public 
```
Then browse `http://localhost:3000`, the page will not show up !

Whereas `bun run dev` works fine !

### Describe the bug

Nuxt generate on a fresh nuxt project does not work anymore.

**Comment by PhE** [comment]
Same result with `npm`.

The page does not show up, `index.html` is missing, the browser shows the folder content:
```
Index of  [public/](http://localhost:3000/)

    [200.html](http://localhost:3000/200.html)
    [404.html](http://localhost:3000/404.html)
    [_nuxt/](http://localhost:3000/_nuxt/)
    [favicon.ico](http://localhost:3000/favicon.ico)
```

**Comment by PhE** [comment]
The problem appears in 3.12.3 but **not in 3.12.2**:

The updated `package.json` : 
```json
{
  "name": "nuxt-app",
  "private": true,
  "type": "module",
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "dependencies": {
    "nuxt": "3.12.2",
    "vue": "latest"
  }
}
```

Then
```
bun install
bun run generate
npx serve .output/public 
```

And browse `http://localhost:3000`, the page will show up !

**Comment by GreyXor** [comment]
Same problem

workaround:
```typescript
nitro: {
  prerender: {
    routes: ["/"],
  },
},
```

**Comment by PhE** [comment]
> Same problem
> 
> workaround:
> 
> ```ts
> nitro: {
>   prerender: {
>     routes: ["/"],
>   },
> },
> ```

Nice workaround, thanks @GreyXor !

Not an issue anymore for me.

I will let the maintainers close this ticket or keep it open.

**Comment by danielroe** [comment]
This should be resolved in the [nightly release](https://nuxt.com/docs/guide/going-further/nightly-release-channel#nightly-release-channel), or in the next release.

Let me know if not and I'll reopen.

**Comment by alcal00** [comment]
> Same problem
> 
> workaround:
> 
> ```ts
> nitro: {
>   prerender: {
>     routes: ["/"],
>   },
> },
> ```

thanks @GreyXor

**Comment by PhE** [comment]
I confirm it is fixed in 3.12.4

No need for the workaround anymore.

**Comment by danielroe** [comment]
Fantastic! 🚀
